twingate/internal/client/client.go

```diff
@@ -61,9 +92,19 @@ func newServerURL(network, url string) serverURL {
 	}
 }
 
+func customRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	// do not retry if API token not set
+	if errors.Is(err, ErrAPITokenNoSet) {
+		return false, err
+	}
+
+	return retryablehttp.DefaultRetryPolicy(ctx, resp, err) //nolint
+}
+
 func NewClient(url string, apiToken string, network string, httpTimeout time.Duration, httpRetryMax int, version string) *Client {
 	sURL := newServerURL(network, url)
 	retryableClient := retryablehttp.NewClient()
+	retryableClient.CheckRetry = customRetryPolicy
```

Comment:
If I understand this code correctly what will happen if I run TF without a Twingate API key is that you'll still make a ton of HTTP requests they will just fail and we ignore that failure?
That's not a good solution...